src/globus_cli/termio/printers/record_printer.py

```diff
@@ -88,6 +97,19 @@ def _key_len(self) -> int:
         """The number of chars in the key column."""
         return max(len(f.name) for f in self._fields) + 2
 
+    @property
+    def _item_wrapper(self) -> TextWrapper:
+        """
+        Access the printers TextWrapper, modifying the width if necessary.
+
+        :returns: a TextWrapper instance for wrapping item values.
+        """
+        # If the class was instantiated with an explicit max_content_width, don't
+        # override it.
+        if not self._is_max_content_width_explicit:
+            self._base_item_wrapper.width = TERM_INFO.columns
+        return self._base_item_wrapper
```

Comment:
I think there's a better path here where `TextWrapper` itself internalizes this setting.

1. Give `TextWrapper` an internal attribute e.g., `TextWrapper._width: int | None = None`
2. Make `TextWrapper.width` a property which sets `_width` to `TERM_INFO.columns` if it is not set
3. Given `TextWrapper.width` a setter which explicitly sets `_width`
4. Get rid of `_item_wrapper` as a property here -- go back to it being an attribute
5. In `__init__`, get rid of `_is_max_content_width_explicit` and instead do the (currently implemented) conditional setting of `width`

I think that solves this and has a less leaky abstraction.

Reply:
Applied; I put it in `terminal_info` so that it's a more general purpose utility but it works the way you described. Nice idea.